presto-hive-common/src/main/java/com/facebook/presto/hive/filesystem/ExtendedFileSystem.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.filesystem;
+
+import com.facebook.presto.hive.HiveFileContext;
+import com.facebook.presto.hive.HiveFileInfo;
+import com.facebook.presto.hive.NamenodeStats;
+import com.facebook.presto.hive.NestedDirectoryPolicy;
+import com.facebook.presto.hive.metastore.Table;
+import com.facebook.presto.hive.util.DirectoryLister;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.LocatedFileStatus;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.PathFilter;
+import org.apache.hadoop.fs.RemoteIterator;
+
+import java.io.IOException;
+import java.util.Iterator;
+
+public abstract class ExtendedFileSystem
+        extends FileSystem
+{
+    public FSDataInputStream openFile(Path path, HiveFileContext hiveFileContext)
+            throws Exception
+    {
+        return open(path);
+    }
+
+    public Iterator<HiveFileInfo> list(
+            DirectoryLister directoryLister,
```

Comment:
Inject `DirectoryLister` to each `HdfsConfiguration` impl (e.g., `HiveCachingHdfsConfiguration`). Then creating the `ExtendedFileSystem` with the injected `DirectoryLister`. That will avoid the param for this interface.

Reply:
I thought we discussed about this. This approach results in to lot of mess. Please remember we need the DirectoryLister for HadoopExtendedFileSystem as well.

Reply:
I haven't compiled or tested yet. But do you think this would work: https://github.com/highker/presto/commit/8784c41240c4ac8e5e6aaa6fe0a737f5b73f5fbc ?

Reply:
It would work. I had started with this approach only.  However I don't like the idea of HdfsEnvironment and PrestoExtendedFileSystemCache dealing with DirectoryLister. It makes Directory lister sit at the core. Also with this too many components, who have nothing to do with list interface, deals with directorylister.

Reply:
That is fine. Having the dirty implementation is better than a confusing interface IMHO. Also, given `DirectoryLister` is already in `presto-hive-common` (which I'm not a big fan frankly), we have already put `DirectoryLister` as a first-class citizen. If there is a way to avoid that, it might be fine with other approaches.

Reply:
James and I discussed and we agreed to switch the list interface to the older way. Basically relying on the directorylister for list functionality.